crates/re_space_view_spatial/src/space_camera_3d.rs

```diff
@@ -52,41 +52,22 @@ impl SpaceCamera3D {
         }
     }
 
-    /// Projects image coordinates into world coordinates
-    pub fn world_from_image(&self) -> Option<Affine3A> {
+    /// Returns x, y, and depth in image/pixel coordinates.
+    pub fn project_onto_2d(&self, point_in_world: Vec3) -> Option<Vec3> {
         let pinhole = self.pinhole?;
-        let world_from_cam = self.world_from_cam();
-        let image_from_cam: Mat3 = pinhole.image_from_cam.into();
-        let cam_from_image = Affine3A::from_mat3(image_from_cam.inverse());
-        Some(world_from_cam * cam_from_image)
-    }
-
-    /// Projects world coordinates onto 2D image coordinates
-    pub fn image_from_world(&self) -> Option<Affine3A> {
-        let pinhole = self.pinhole?;
-        let cam_from_world = self.cam_from_world();
-
-        let image_from_cam = pinhole.image_from_cam.into();
-        let image_from_cam = Affine3A::from_mat3(image_from_cam);
-        Some(image_from_cam * cam_from_world)
-    }
```

Comment:
These two functions were just big footguns. These were not affine transforms, nor even normal 4x4 projection matrices. Instead they where matrices with projection in the Z component (not W), so just… wrong. Better to be explicit than try to combine two very different kind of transforms like this.